Game/space-invaders/src/Game.cpp

```diff
@@ -27,30 +26,91 @@ void Game::Init()
     m_SpriteRenderer = std::make_unique<SpriteRenderer>(spriteShader);
     
     // Create Player
-    m_PlayerManager = std::make_unique<PlayerManager>();
+    m_PlayerManager = std::make_unique<PlayerManager>(*this);
     m_PlayerManager->CreatePlayer(WIDTH, HEIGHT);
 }
 
 void Game::Update(float deltaTime)
 {
-    // std::cout << "Game running\n";
+    // move player projectiles
+    for (auto& projectile : m_PlayerProjectiles)
+    {
+        // check boundaries
+        if (projectile.Position.y <= 0)
```

Comment:
I think it's better to update the position and then check for the out of screen, otherwise it will check for it only on the next frame